syncopy/io/_load_nwb.py

```diff
@@ -202,61 +220,63 @@ def load_nwb(filename, memuse=3000):
+        # allocate a target dataset for reading the NWB data
+        if container is not None:
+            filename = filebase + "_" + acqValue.name + ".analog"
+        else:
+            filename = None
+            
+        angData = AnalogData(dimord=AnalogData._defaultDimord, filename=filename)
+        angShape = [None, None]
+        angShape[angData.dimord.index("time")] = acqValue.data.shape[0]
+        angShape[angData.dimord.index("channel")] = acqValue.data.shape[1]
+        h5ang = h5py.File(angData.filename, mode="w")
+        angDset = h5ang.create_dataset("data", dtype=np.result_type(*dTypes), shape=angShape)
+        
+        # If channel-specific gains are set, load them now
+        if acqValue.channel_conversion is not None:
+            gains = acqValue.channel_conversion[()]
+            if np.all(gains ==  gains[0]):
+                gains = gains[0]
+                
         # Given memory cap, compute how many data blocks can be grabbed per swipe:
         # `nSamp` is the no. of samples that can be loaded into memory without exceeding `memuse`
         # `rem` is the no. of remaining samples, s. t. ``nSamp + rem = angDset.shape[0]`
         # `blockList` is a list of samples to load per swipe, i.e., `[nSamp, nSamp, ..., rem]`
-        nSamp = int(memuse / (np.prod(angDset.shape[1:]) * angDset.dtype.itemsize))
+        nSamp = int(memuse / (acqValue.data.shape[1] * angDset.dtype.itemsize))
         rem = int(angDset.shape[0] % nSamp)
         blockList = [nSamp] * int(angDset.shape[0] // nSamp) + [rem] * int(rem > 0)
-
-        # If channel-specific gains are set, load them now
-        if acqValue.channel_conversion is not None:
-            gains = acqValue.channel_conversion[()]
-            
-        endChan = chanCounter + acqValue.data.shape[1]
+        
         for m, M in enumerate(tqdm(blockList, desc=pbarDesc, position=1, leave=False)):
             st_samp, end_samp = m * nSamp, m * nSamp + M
-            angDset[st_samp : end_samp, chanCounter : endChan] = acqValue.data[st_samp : end_samp, :]
+            angDset[st_samp : end_samp, :] = acqValue.data[st_samp : end_samp, :]
             if acqValue.channel_conversion is not None:
-                angDset[st_samp : end_samp, chanCounter : endChan] *= gains
-
-        # Update channel counter for next `acqValue``
-        chanCounter += acqValue.data.shape[1]
-
-    # Finalize angData
-    angData.data = angDset
-    angData.channel = chanNames
-    angData.samplerate = sRates[0]
-    angData.trialdefinition = trl
-    angData.info = {'starting_time' : tStarts[0]}
-
-    # Write logs
-    msg = "Read data from NWB file {}".format(nwbFullName)
-    angData.log = msg
-    if evtData is not None:
-        evtData.log = msg
-
-    return angData, evtData
+                angDset[st_samp : end_samp, :] *= gains
+            
+        # Finalize angData
+        angData.data = angDset
+        angData.channel = acqValue.electrodes[:].location.to_list()
```

Comment:
Channels again: if channel names are not defined for the current `acquisition`, then this might file, I think.

Reply:
Ah yes true, we should add back in your previous check there right?

```
if channels.unique().size == 1:
    SPYWarning("No channel names found for {}".format(acqName))
else:
    angData.channel = acqValue.electrodes[:].location.to_list()
```

Reply:
Yes, I think this should do it :)